src/services/activityReports.js

```diff
@@ -184,7 +186,30 @@ async function saveReportRecipients(
     };
   }
 
-  await ActivityRecipient.bulkCreate(newRecipients, { ignoreDuplicates: true });
+  await Promise.all(
+    newRecipients.map(async (newRecipient) => {
+      if (newRecipient.grantId) {
+        return ActivityRecipient.findOrCreate({
+          where: {
+            activityReportId: newRecipient.activityReportId,
+            grantId: newRecipient.grantId,
+          },
+          defaults: newRecipient,
+        });
+      }
+      if (newRecipient.otherEntityId) {
+        return ActivityRecipient.findOrCreate({
+          where: {
+            activityReportId: newRecipient.activityReportId,
+            otherEntityId: newRecipient.otherEntityId,
+          },
+          defaults: newRecipient,
+        });
+      }
+      return null;
```

Comment:
We will want to remove nulls, with a filter before the map looking for recipients with no `grantId` or `otherEntityId` or after looking for nulls